### PR TITLE
[#12048] Migrate Session Links Recovery Action

### PR DESCRIPTION
**Pull request by NicolasCwy**
Part of #12048 

**Outline of Solution**
Using the refactored email generator (#12723), the links of feedback sessions which students are in are fetched from data store and Postgres server and are combined into a single email to be sent to students (this should be removed after migration)

### CODE REVIEW

File: src/it/java/teammates/it/sqllogic/api/EmailGeneratorTestIT.java [hunk 1]
```diff
@@ -106,12 +113,14 @@ public void testGenerateSessionLinksRecoveryEmail() throws Exception {
...
         email = emailGenerator.generateSessionLinksRecoveryEmailForStudent(
-                student1InCourse4.getEmail());
+                student1InCourse4.getEmail(), nonExistentStudent, emptyFragmentList);
 
         subject = EmailType.SESSION_LINKS_RECOVERY.getSubject();
 
         verifyEmail(email, student1InCourse4.getEmail(), subject,
                 "/sessionLinksRecoveryOpenedOrClosedAndpublishedSessions.html");
+
+        ______TS("Migration test: students from both postgres and datastore are fetched");
```
Comment (weiquu) [inline src/it/java/teammates/it/sqllogic/api/EmailGeneratorTestIT.java:123]: Are we doing this here? If not I think can remove first

**Review by weiquu — APPROVED** [review]
LGTM! Just quick check of the migration test below

File: src/it/java/teammates/it/sqllogic/api/EmailGeneratorTestIT.java (hunk 1 above)
Comment (NicolasCwy) [reply src/it/java/teammates/it/sqllogic/api/EmailGeneratorTestIT.java]: Yup my bad! Didn't want to mock datastore values here, so I'll add it as a unit test in another PR

**Review by cedricongjh — APPROVED** [review]
LGTM, thanks!

**Pull request merged**